ycmd.el

```diff
@@ -260,6 +260,24 @@ use `ycmd-parse-buffer'."
   :group 'ycmd
   :type '(alist :key-type symbol :value-type (repeat string)))
 
+(defconst ycmd-semantic-completion-file-types
+  '("c"
+    "cpp"
+    "objc"
+    "objcpp"
+    "cs"
+    "python")
+  "A list of ycmd file type strings which support semantic completion.")
+
+(defun ycmd-semantic-file-types (mode)
+  "Find the ycmd file types for MODE which support semantic completion.
+
+Returns a possibly empty list of ycmd file type strings.  If this
+is empty, then ycmd doesn't support semantic completion (or
+diagnostics) for MODE."
+  (let ((file-types (assoc-default mode ycmd-file-type-map)))
+    (-intersection file-types ycmd-semantic-completion-file-types)))
```

Comment:
Why do you return the filetype string? Wouldn't it be sufficient to make the function a predicate function `ycmd-semantic-file-type-p` and use `memq` instead of `-intersection`?

Or do you want to use this function for more than checking if the current mode supports semantic file checks?